posthog/temporal/tests/batch_exports/test_s3_batch_export_workflow.py

```diff
@@ -251,6 +251,14 @@ async def assert_clickhouse_records_in_s3(
     if batch_export_model is not None:
         if isinstance(batch_export_model, BatchExportModel):
             batch_export_schema = batch_export_model.schema
+            if batch_export_model.filters is not None:
+                exclude_without_properties = [f"test-no-prop-{i}" for i in range(5)]
+
+                if not exclude_events:
+                    exclude_events = exclude_without_properties
+                else:
+                    exclude_events.extend(exclude_without_properties)
+
```

Comment:
not quite sure I follow these changes

Reply:
It's a bit of a hack. The model I've added to the test only exports events matching these properties:
```
        filters=[
            {"key": "$browser", "operator": "exact", "type": "event", "value": ["Chrome"]},
            {"key": "$os", "operator": "exact", "type": "event", "value": ["Mac OS X"]},
        ],
```. But in our tests we always create 5 events without properties, named "test-no-prop-{i}". These events will not be exported when using the model with filters (Which is expected), so we exclude them from the result by adding them to `exclude_events`.

Reply:
Maybe we should just pass the filters along to `iter_records`. I just wanted a quick way of testing the changes.

Reply:
yeah, if not too much effort I would say it's worth adding to `iter_records` as it would make things a bit clearer and I can imagine some confusion in future when certain tests start failing and need to figure out why

Reply:
Yeah, I've updated `iter_records` and the test code to pass the filters along.

Reply:
This means a lot of views can be removed from batch_exports.py, as those were copies only used in tests.